Description: Restrict NTL_zz_pX::init to Coeff vector
 The previous template allowed initializing a polynomial from any
 vector type, but on some platforms (e.g., i386) this caused ambiguous
 conversions when E is Givaro::Integer, because NTL::SetCoeff expects
 a long. This patch removes the generic template overload and keeps
 the overload specifically for std::vector<Coeff>, avoiding type
 conversion errors.
Author: Jean-Guillaume Dumas <Jean-Guillaume.Dumas@imag.fr>
Origin: https://github.com/linbox-team/linbox/commit/9555c5b
Bug: https://github.com/linbox-team/linbox/issues/328
Last-Update: 2025-09-11


--- a/linbox/ring/ntl/ntl-lzz_px.h
+++ b/linbox/ring/ntl/ntl-lzz_px.h
@@ -191,9 +191,7 @@
 		 * of the vector corresponds to the leading coefficients.  That is,
 		 * v[i] = coefficient of x^i.
 		 */
-		//Element& init( Element& p, const std::vector<Coeff>& v ) const
-      template <class E>
-		Element& init( Element& p, const std::vector<E>& v ) const
+		Element& init( Element& p, const std::vector<Coeff>& v ) const
 		{
 			p = 0;
 			for( long i = 0; i < (long)v.size(); ++i )
